inc/namespace.php

```diff
@@ -1823,3 +1826,33 @@ function handle_autosuggest_endpoint() {
 	// Return JSON response.
 	wp_send_json( $data, 200 );
 }
+
+/**
+ * Limit search query string length.
+ *
+ * Prevents overly complex queries that can be slow to process and tie up resources.
+ *
+ * @param WP_Query $query The WP_Query instance (passed by reference).
+ */
+function limit_search_query_length( WP_Query $query ) : void {
+	if ( empty( $query->get( 's' ) ) ) {
+		return;
+	}
+
+	$search = $query->get( 's' );
+
+	// Get max length from config.
+	$max_length = Altis\get_config()['search']['max-query-length'] ?? 80;
+
+	/**
+	 * Filters the maximum allowed query string length for searches.
+	 *
+	 * @param int $max_length The maximum allowed query string length.
+	 */
+	$max_length = (int) apply_filters( 'altis.search.max_query_length', $max_length );
```

Comment:
To @ntwb's point, we could pass the `$search` value into the filter here, so you could filter this based on the search itself, e.g. if the `$search` looked like a URL, update the `$max_length` to some suitable large number, otherwise keep the default behavior.

```suggestion
	$max_length = (int) apply_filters( 'altis.search.max_query_length', $max_length, $search );
```